src/nvim/event/proc.c

```diff
@@ -449,8 +449,9 @@ static void on_proc_exit(Proc *proc)
         && proc == &server_chan->stream.proc) {
       // Need to call ui_client_may_restart_server() here as well, as sometimes
       // rpc_close_event() hasn't been called yet (also see comments above).
+      bool connected = ui_client_may_connect();
       ui_client_may_restart_server();
```

Comment:
if connected=true we would never want to call `ui_client_may_restart_server`, right?

not a blocker for this pr, but just a reminder